api/internal/managers/app/release/template.go

```diff
@@ -63,6 +66,77 @@ func (m *appReleaseManager) TemplateHelmChartCRs(ctx context.Context, configValu
 	return templatedCRs, nil
 }
 
+// DryRunHelmChart finds the corresponding chart archive and performs a dry run templating of a Helm chart
+func (m *appReleaseManager) DryRunHelmChart(ctx context.Context, templatedCR *kotsv1beta2.HelmChart) ([][]byte, error) {
+	if templatedCR == nil {
+		return nil, fmt.Errorf("templated CR is nil")
+	}
+
+	if m.releaseData == nil {
+		return nil, fmt.Errorf("release data not initialized")
+	}
```

Comment:
Thoughts on moving the `releaseData` checks to the `NewAppReleaseManager` and do it only once instead (for all the methods)?

Reply:
Will do in a follow up PR. Thanks!